pkg/ssh/manager.go

```diff
@@ -39,10 +57,11 @@ func (sm *StreamManager) NewStreamHandler(
 	writeC := make(chan *extensions_ssh.ServerMessage, 32)
 	sh := &StreamHandler{
 		auth:       auth,
-		config:     cfg,
+		config:     sm.cfg,
 		downstream: downstream,
 		readC:      make(chan *extensions_ssh.ClientMessage, 32),
 		writeC:     writeC,
+		reauthC:    make(chan struct{}),
```

Comment:
This channel needs a buffer size of 1, otherwise reauth will be skipped if the stream is in the middle of handling a message